src/App.tsx

```diff
@@ -17,18 +55,33 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setFilteredSelect={setFilteredSelect}
+                query={query}
+                setQuery={setQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {visibleTodos.length === 0 && <Loader />}
```

Comment:
will be better to create separate variable for defining was data loaded or no
for now if you'll receive an empty array from the server, user will see infinite loader